internal/policy/policy.go

```diff
@@ -278,49 +288,57 @@ func parseEffectiveTime(choosenTime string) (*time.Time, error) {
 
 // checkOpts returns an instance based on attributes of the Policy.
 func checkOpts(ctx context.Context, p *policy) (*cosign.CheckOpts, error) {
+	var err error
 	opts := cosign.CheckOpts{}
 
-	verifier, err := signatureVerifier(ctx, p)
-	if err != nil {
-		return nil, err
-	}
-	opts.SigVerifier = verifier
-
 	if p.RekorUrl == "" {
+		// TODO: Rethink this? Maybe have an explicit flag to do ignore rekor.
 		opts.IgnoreTlog = true
+		log.Debug("Transparency log verification disabled")
 	} else {
-		rekorClient, err := rekor.NewClient(p.RekorUrl)
-		if err != nil {
+		if opts.RekorClient, err = rekor.NewClient(p.RekorUrl); err != nil {
 			log.Debugf("Problem creating a rekor client using url %q", p.RekorUrl)
 			return nil, err
 		}
-
-		opts.RekorClient = rekorClient
 		log.Debug("Rekor client created")
 
-		// TODO the Rekor public key and entry id should originate in the policy
-		rekorPublicKeyPEM := os.Getenv("REKOR_PUBLIC_KEY")
-		if rekorPublicKeyPEM != "" {
-			rekorPublicKey, err := cryptoutils.UnmarshalPEMToPublicKey([]byte(rekorPublicKeyPEM))
-			if err != nil {
-				return nil, err
-			}
-			rekorPublicKeyBytes, err := x509.MarshalPKIXPublicKey(rekorPublicKey)
-			if err != nil {
-				return nil, err
-			}
-			digest := sha256.Sum256(rekorPublicKeyBytes)
-			logId := hex.EncodeToString(digest[:])
-
-			opts.RekorPubKeys = &cosign.TrustedTransparencyLogPubKeys{
-				Keys: map[string]cosign.TransparencyLogPubKey{
-					logId: {
-						PubKey: rekorPublicKey,
-						Status: tuf.Active,
-					},
-				},
-			}
+		if opts.RekorPubKeys, err = cosign.GetRekorPubs(ctx); err != nil {
+			return nil, err
+		}
+		log.Debug("Retrieved Rekor public keys")
```

Comment:
We could pull this change as a separate PR and have it merged in main... Does it get the key from TUF or from Rekor?

Reply:
+1, and yes it does get it from TUF.